dataclients/kubernetes/testdata/ingress/eastwestrange/ing-with-1externalhost-1internalhost-1prule-1customroute-2ranges-1match.eskip

```diff
@@ -1,17 +1,17 @@
 kube_foo__qux__0__example_ingress_cluster_local_____:
-	Host("^example[.]ingress[.]cluster[.]local$") && PathRegexp("^/")
+	Host("^(example[.]ingress[.]cluster[.]local(:[0-9]+)?)$") && PathRegexp("^/")
```

Comment:
Why do we need nested capture groups here?
I think it should be `^example[.]ingress[.]cluster[.]local(:[0-9]+)?$"`, no?

Reply:
The outer brackets are added by the regexp generator function that supports a list of multiple hosts:
https://github.com/zalando/skipper/blob/ec6e5c7246d4f33b67a051e0fb80342af2061ae1/dataclients/kubernetes/hosts.go#L11-L22
It is possible to omit the outer brackets by checking the list length: 
```diff
+	 	if len(hrx) == 1 {
+	 		return "^" + hrx[0] + "$"
+	 	}
```

The values of capture groups are not used as far as I can tell so it does not really matter I think (we could also consider using non capturing groups for purity but that would look even more clumsy like `^(?:example[.]ingress[.]cluster[.]local(?::[0-9]+)?)$`). See also [subject regexp in the playground](https://regex101.com/r/Kljl0Y/1).

Reply:
The outer brackets are needed, without them in case of multiple hosts [a prefixed value will match](https://regex101.com/r/sa90ZK/1):
`^example[.]org(:[0-9]+)?|example[.]org(:[0-9]+)?$` matches `www.example.org`

Reply:
Without outer brackets [a prefixed value matches](https://regex101.com/r/sa90ZK/3):
`^example[.]org(:[0-9]+)?|example[.]com(:[0-9]+)?$` matches `www.example.com`.

Alternatively every host could be wrapped with `^...$` like [`^example[.]org(:[0-9]+)?$|^example[.]com(:[0-9]+)?$`](https://regex101.com/r/kUVIxl/1)